lib/auth/in_memory/checker.js

```diff
@@ -0,0 +1,173 @@
+ *                             what we're checking
+ * @param {string} name - the name of the entity we're checking for
+ * @param {string} type - expected typename of the entity we're checking
+ * @param {object} obj - the object we're checking the fields of
+ * @return {boolean} true if the field exists and type is Ok
+ *                   false if an error was found and reported
+ */
+function checkExists(data, container, name, type, obj) {
+    if (obj[name] === undefined) {
+        data.errors.push({
+            txt: 'missing property in auth entity',
+            obj: {
+                entity: container,
+                property: name,
+            },
+        });
+        return false;
+    }
+    return checkType(data, container, name, type, obj);
+}
+
+function checkUser(data, userObj) {
+    if (checkExists(data, 'User', 'arn', 'string', userObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.arns[userObj.arn] = incr(data.arns[userObj.arn]);
+    }
+    if (checkExists(data, 'User', 'email', 'string', userObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.emails[userObj.email] = incr(data.emails[userObj.email]);
+    }
+    checkExists(data, 'User', 'keys', 'array', userObj);
+
+    if (userObj.keys) {
+        if (checkType(data, 'User', 'keys', 'array', userObj)) {
+            userObj.keys.forEach(keyObj => {
+                // eslint-disable-next-line no-param-reassign
+                data.keys[keyObj.access] = incr(data.keys[keyObj.access]);
+            });
+        }
+    }
+}
+
+function checkAccount(data, accountObj) {
+    if (checkExists(data, 'Account', 'email', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.emails[accountObj.email] = incr(data.emails[accountObj.email]);
+    }
+    if (checkExists(data, 'Account', 'arn', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.emails[accountObj.arn] = incr(data.emails[accountObj.arn]);
+    }
+    if (checkExists(data, 'Account', 'canonicalID', 'string', accountObj)) {
+        // eslint-disable-next-line no-param-reassign
+        data.emails[accountObj.canonicalID] =
+            incr(data.emails[accountObj.canonicalID]);
+    }
+
+    if (accountObj.users) {
+        if (checkType(data, 'Account', 'users', 'object', accountObj)) {
+            accountObj.users.forEach(userObj => checkUser(data, userObj));
+        }
+    }
+
+    if (accountObj.keys) {
+        if (checkType(data, 'Account', 'keys', 'object', accountObj)) {
+            accountObj.keys.forEach(keyObj => {
+                // eslint-disable-next-line no-param-reassign
+                data.keys[keyObj.access] = incr(data.keys[keyObj.access]);
+            });
+        }
+    }
+}
+
+function dumpCountError(property, obj) {
+    Object.keys(obj).forEach(key => {
+        if (obj[key] > 1) {
+            log.error('property should be unique', {
```

Comment:
Which is a good news, but there's still the point of how much can the config object be used, if it's incompletely built...
I'll test it anyways ^^' Will keep you posted